packages/api/src/brevo-api/brevo-api-campaigns.service.ts

```diff
@@ -14,7 +15,11 @@ export class BrevoApiCampaignsService {
 
     constructor(@Inject(BREVO_MODULE_CONFIG) private readonly config: BrevoModuleConfig) {
         this.campaignsApi = new SibApiV3Sdk.EmailCampaignsApi();
-        this.campaignsApi.setApiKey(SibApiV3Sdk.EmailCampaignsApiApiKeys.apiKey, config.brevo.apiKey);
+    }
+
+    private setApiKey(scope: EmailCampaignScopeInterface): void {
+        const { apiKey } = this.config.brevo.getBrevoConfig(scope);
+        this.campaignsApi.setApiKey(SibApiV3Sdk.EmailCampaignsApiApiKeys.apiKey, apiKey);
```

Comment:
I believe this isn't a good idea: One could forget to call the the setApiKey-method, and as the service isn't request-scoped, the API key of the previous request will be used. A better approach would be to remove the campaignsApi member and add a `getCampaignsApi(scope)` method that creates a new campaign API for the scope (the result could be cached in a map).

Reply:
I haven't thought about that, this could be really bad. I changed it to your suggested approach: 
Removing the member: [bea2fd3](https://github.com/vivid-planet/comet-brevo-module/pull/53/commits/bea2fd3715bbd88853b4ce66d8f15c1870a1018c)
Adding a map to store the created apis for the scope: [8b1ecf2](https://github.com/vivid-planet/comet-brevo-module/pull/53/commits/8b1ecf231b75a7340b6bcf76461240cc32887b6d)